src/app/src/pages/Login.js

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { Box, Button, Input, Text, VStack } from '@chakra-ui/react';
+
+import { API } from '../api';
+import { API_URLS } from '../constants';
+import { login } from '../store/authSlice';
+
+export default function Login({ to }) {
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [errorDetail, setErrorDetail] = useState('');
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+
+    const loginRequest = () => {
+        API.post(API_URLS.LOGIN, {
+            email,
+            password,
+        })
+            .then(() => {
+                dispatch(login());
+            })
+            .catch(apiError => {
+                setErrorDetail(apiError.response?.data?.detail);
+            });
+    };
+
+    const signedIn = useSelector(state => state.auth.signedIn);
+
+    // Upon successful sign in, redirect if specified (e.g. by /login route)
+    useEffect(() => {
+        if (signedIn && to) {
+            navigate(to, { replace: true });
+        }
+    }, [navigate, to, signedIn]);
+
+    return (
+        <VStack bg='gray.100' h='100vh' spacing={5}>
+            <Box h='150px'></Box>
```

Comment:
Rather than having a fixed gap in the top, prefer centering to allow for different screen sizes:

```suggestion
        <VStack bg='gray.100' h='100vh' spacing={5} justify='center'>
```

Reply:
Could also do something like:

```jsx
<Box h='15%' />
```

Reply:
I like the second approach; I didn't think the original design was centered. d974738dd8dfec7a445befed3c0738182ce97487